pipelines/trapum_search_pipeline/webpage_riptide_pipeline_ddplan_pulsarx.py

```diff
@@ -0,0 +1,261 @@
+
+async def riptide(config_file, rootname, processing_dir):
+    
+    #Generate riptide config file
+    log.info("Generating riptide config file")
+    config_file_path = processing_dir+'config.yaml'
+    config_file_handle = open(config_file_path, 'w')
+    config_file_handle.write(config_file)
+    config_file_handle.close()
+    
+    cmd = (f"rffa -k {config_file_path}  "
+           f"-o {rootname} *.inf")
+    
+    try:
+        await shell_call(cmd, cwd=processing_dir)
+    except Exception as error:
+        log.error("riptide call failed, cleaning up partial files")
+        delete_files_if_exists(processing_dir)
+        raise error
+
+#%% Pipeline
+
+async def riptide_pipeline(data, status_callback):
+    # Limit core size to avoid ephemeral-storage overflows
+    # in the event of segmentation faults
+    await shell_call("ulimit -c  0")
+
+    #Get processing args
+    processing_args = data["processing_args"]
+    output_dir = data["base_output_dir"]
+    processing_id = data["processing_id"]
+    config_file = processing_args["config_file"]
+    rfi_flags = processing_args["rfi_flags"]
+    ddplan_args = processing_args["ddplan"]
+
+    log.info(f"Creating output directory: {output_dir}")
+    os.makedirs(output_dir, exist_ok=True)
+    output_dps = []
+
+    for pointing in data["data"]["pointings"]:
+        for beam in pointing["beams"]:
+            
+            beam_name = beam["name"]
+            
+            dps = sorted(select_data_products(
+                beam, lambda fname: fname.endswith(".fil")))
+
+            if processing_args["temp_filesystem"] == "/beeond/":
+                log.info("Running on Beeond")
+                processing_dir = os.path.join(BEEOND_TEMP_DIR, str(processing_id))
+            else:
+                log.info("Running on BeeGFS")
+                processing_dir = os.path.join(output_dir, "processing/")
+            os.makedirs(processing_dir, exist_ok=True)
+            
+            try:
+                #Dedisperse the filterbanks and output to temporary processing directory
+                log.info(
+                    "Dedispersing filterbanks and performing RFI mitigation")
+                status_callback(
+                    "Dedispersing filterbanks and performing RFI mitigation")
+                await dedisperse_all_fil(dps, processing_dir,
+                               f"dedispersed_{processing_id}_",
+                              ddplan_args, beam_name,
+                              rfi_flags)
+                
+                #Run riptide in the temporary processing directory
+                log.info("Riptide search")
+                status_callback("Riptide search")
+                await riptide(config_file,
+                    f"riptide_{processing_id}_", processing_dir)
+                
+                #Remove unnecessary riptide files
+                log.info("Removing unnecessary riptide files")
+                os.remove(processing_dir+'/*.json')
```

Comment:
I don't think os.remove works with wildcards. Have you tried this?

Reply:
Thanks. Should be fixed in new commit.